lib/rubyoctopus/octopusconnection.rb

```diff
@@ -20,8 +20,15 @@ def initialize(url, api_key)
       end
     end
 
-    def get(resource_type)
-      @conn.get(resource_type)
+    # Makes a GET request to the given URL with the given query parameters.
+    # @param url [String] The URL to be appended on the base URL for the GET request.
+    # @param query_parameters [Hash<String, String>] A hash of query parameters for the request.
+    def get(url, query_parameters = nil)
```

Comment:
What about making the default `query_parameters` value `{}` so you don't need a nil check here?

Reply:
You can see how I did it in the resource paging branch [here](https://github.com/Ruby-Octopus/ruby-octopus/blob/resource-paging/lib/rubyoctopus/octopusconnection.rb).

Reply:
Definitely prefer your comments, though!

Reply:
I thought about using `{}` as the default value, but I think Faraday didn't like me passing in an empty object (or at least it made your test fail). Maybe I could've just tweaked your test though 🤷 .

Oh, I didn't look at any of your work-in-progress branches; didn't realize you'd already done this.

Reply:
If it's a problem with the test, it could easily be because the specification for the method changed.

Reply:
Yeah, the test works fine if you update it to say `.with(TEST_PATH, {})`.